common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockPusher.java

```diff
@@ -77,42 +80,55 @@ public OneForOneBlockPusher(
 
     @Override
     public void onSuccess(ByteBuffer response) {
-      // On receipt of a successful block push
-      listener.onBlockPushSuccess(blockId, new NioManagedBuffer(ByteBuffer.allocate(0)));
+      // If the response is non-empty, the server has responded some error code. Decode
+      // it and handle the error accordingly.
+      if (response.hasRemaining()) {
```

Comment:
Shall we add a new code for the successful response? I think that would be more reboust. We can rename `PushBlockNonFatalErrorCode` to `BlockPushResponse` if we go this way.

Reply:
I saw you mentioned the cost of adding extra data for each success push. But I think it's should be fine since the status code only takes 1 byte. Even with the empty body, the `RpcResponse` itself would take some bytes. 1 byte is small compares to the latter.

Reply:
You are right that we would anyway incur the overhead with decoding an `RpcResponse` with every success response even if the body is empty.
Agree that having a dedicated code for success push makes it more consistent.